src/gui/properties/speedplotview.h

```diff
@@ -117,13 +116,13 @@ class SpeedPlotView : public QGraphicsView
     boost::circular_buffer<PointData> m_data5Min;
     boost::circular_buffer<PointData> m_data30Min;
     boost::circular_buffer<PointData> m_data6Hour;
+    Averager m_averager30Min;
```

Comment:
How about moving `m_data30Min` into `m_averager30Min`?
This would simplify the objects relations.

Reply:
Hmm, maybe it is just me that didn't expect to see such complications for such simple thing...

Reply:
Current design is acceptable for me. It can be changed, but it will require more changes than you think. So let it be as is.